python/tank/descriptor/io_descriptor/git_branch.py

```diff
@@ -123,24 +123,23 @@ def download_local(self):
             # nothing to do!
             return
 
-        # cache into the primary location
+        # cache into the temporary location
+        temporary_path = self._get_temporary_cache_path()
         target = self._get_primary_cache_path()
-
         try:
             # clone the repo, switch to the given branch
             # then reset to the given commit
             commands = [
                 "checkout -q \"%s\"" % self._branch,
                 "reset --hard -q \"%s\"" % self._version
             ]
-            self._clone_then_execute_git_commands(target, commands)
-
+            self._clone_then_execute_git_commands(temporary_path, commands)
         except Exception, e:
             raise TankDescriptorError(
```

Comment:
Git descriptors actually don't clean up on a failed clone, so the temporary download will remain.